docs/contents/advanced_usage.md

```diff
@@ -0,0 +1,223 @@
+11.5), you can edit the `setup.py` to specify that you are *not* compatible
+with the latest of path.py:
+
+```python
+setup(
+    ...
+    install_requires=[
+      "path.py < 11.5",
+    ],
+)
+```
+
+## Freeze dev dependencies
+
+The above approach does not work really well if you use a linter like `pylint`
+or `flake8`, of even a type checker like `mypy`
+
+This is because new releases of those tools often cause new warnings or errors
+to be produced, so you only want to update them when you're ready.
+
+Thus, a good practice is to freeze the versions of those tools directly in the
+`setup.py`:
+
+```python
+setup(
+
+  extras_require={
+        "dev": [
+            "flake8==3.5.0",
+        ]
+  }
+
+)
+```
+
+That way you can freely re-run `dmenv lock`, even in a completely fresh
+environment.
+
+
+## Upgrading just one development dependency
+
+For instance if there's a bug in `py`, you can bump `py` version by editing the
+lock file directly:
+
+```patch
+- py==1.6.0
++ py==1.7.0
+```
+
+## Upgrading just one regular dependency
+
+If the bug is in one of the concrete dependencies, you should update the `setup.py` file instead
+
+```patch
+    install_requires=[
+      "path.py"
++     "importlib-metadata >= 0.6"
+  ]
+```
+
+That way consumers of your code *will* get the correct version.
+
+Then run `dmenv lock` without cleaning the virtual environment so that
+`importlib-metadata` gets upgraded and its new version frozen.
+
+
+## Using dependencies from git URLs
+
+Let's say you came across a bug that's only fixed on the `master` branch of
+`pytest` on GitHub, specifically at the commit `deadbeef`.
+
+One solution is to replace the line in `requirements.lock` to use a git URL
+like this:
+
+```text
+# requirements
+git+https://github.com/pytest-dev/pytest@deadbeef#egg=pytest
+```
+
+In that case, `pip freeze` will contain a line looking like `pytest==4.0b1`, where `4.0b1`
+is the `pytest` version at this particular commit.
+
+When it comes to re-generating the lock, `dmenv` will see that there is already a line
+specifying `pytest` version in a more precise manner, so it will keep the `git` line
+in the lock and ignore the non-precise `4.0b1` version.
+
+
+## Using different system platforms
+
+Sometimes a concrete dependency will only be available on a specify platform.
+
+So if you've generated the lock file on Linux, you may get different results on Windows.
+
+One way to solve this is to run `dmenv lock` with the `--platfrom` argument.
+
+Existing lines in the lock file will be kept, and any *new* dependency will be suffixed
+with a [platform marker](https://www.python.org/dev/peps/pep-0508/), like this:
+
+```text
+# requirements.lock, generated on liux
+foo==0.2
+
+> dmenv lock --platform windows   # run on Windows
+
+# requirements.lock
+foo==0.2
+pywin2==0.42 ; platform == "windows"
+```
+
+## Using different Python versions
+
+If you want your cod to be run across different Python versions, you may encounter similar issues.
```

Comment:
I assume you mean "code", and not this beautiful creature =]

![cod](https://user-images.githubusercontent.com/5155966/51984722-4a350e80-249c-11e9-9673-f93deb7ff9e9.jpg)